src/components/comment-input.tsx

```diff
@@ -20,11 +22,31 @@ const InputContainer = styled.div`
 `;
 
 export const CommentInput: React.VoidFunctionComponent = () => {
+  const { setNewCommentText, newCommentText, comments, setComments } =
+    useComments();
+
+  const onSend = () => {
+    const comment: Comment = {
+      id: comments.length,
+      user: "anon",
+      text: newCommentText,
+      date: new Date().toUTCString(),
+    };
+    setComments([...comments, comment]);
+    setNewCommentText("");
+  };
+
   return (
     <InputContainer>
       <AvatarImage />
-      <TextInput placeholder="Добавить комментарий..." />
-      <Button>Отправить</Button>
+      <TextInput
+        placeholder="Добавить комментарий..."
+        value={newCommentText}
+        onChange={(e: ChangeEvent<HTMLInputElement>) =>
+          setNewCommentText(e.target.value)
+        }
+      />
+      <Button onClick={() => onSend()}>Отправить</Button>
```

Comment:
As a general rule, we can try avoiding inline functions in JSX and go for defining `handleXyz` or using `onXyz` directly if that’s possible. This saves us from long inline functions in the middle of JSX, which make it harder to imagine the resulting three.

Avoiding inline functions in JSX also allows us to wrap handlers into `React.useCallback` which may be helpful to avoid unnecessary re-renders of subtrees. This would be a micro-optimisation at this point, so the main benefit is improving readability by sticking with consistent code rhythm.